### ISSUE TITLE
user manager creates empty cgroups for system mount and swap units

### ISSUE THREAD

**Issue by keszybz** [systemd/systemd#14475]
```console
$ find /sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/ -name '*.procs' -exec grep -L . {} \;|grep -vF '.slice/cgroup.procs' 
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/var-tmp-systemd\x2dtest.XKpA2f-root.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/sys-fs-fuse-connections.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/boot.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/dev-dm\x2d2.swap/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/sys-kernel-config.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/sys-kernel-debug.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/home.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/var-lib-snapd-snap-fedora29-x1.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/dev-zram0.swap/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/var-lib-snapd-snap-core-7270.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/dev-disk-by\x2did-dm\x2dname\x2dfedora_krowka\x2dswap.swap/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/-.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/var-lib-snapd-snap-hello\x2dfedora-4.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/run-user-6.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/dbus\x2d:1.2\x2dorg.gnome.ControlCenter.SearchProvider.slice/dbus-:1.2-org.gnome.ControlCenter.SearchProvider@6.service/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/pipewire.socket/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/dev-mqueue.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/var-tmp-systemd\x2dtest.uzaGqr-root.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/dbus.socket/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/var-lib-snapd-snap-powershell\x2dpreview-25.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/var-tmp-systemd\x2dtest.E8gbhM-root.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/tmp.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/var-lib-snapd-snap-core-7396.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/boot-efi.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/var-tmp-systemd\x2dtest.t5lo01-root.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/dev-disk-by\x2duuid-12ba0393\x2dce3c\x2d4e08\x2d8571\x2d515e9c20aab6.swap/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/run-user-1000.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/sys-kernel-debug-tracing.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/dev-fedora_krowka-swap.swap/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/var-tmp-systemd\x2dtest.9BNxB3-root.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/proc-sys-fs-binfmt_misc.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/var-tmp-systemd\x2dtest.KOJLim-root.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/pulseaudio.socket/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/var-lib-snapd-snap-powershell\x2dpreview-26.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/dev-disk-by\x2did-dm\x2duuid\x2dLVM\x2doPw3E5e3dNDxykLU002jdZQYALnpkXfEK5SqKEM0dHJV57eT04xRkaMjgevXWvR2.swap/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/dev-hugepages.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/var-tmp-systemd\x2dtest.FaQYKQ-root.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/var-lib-nfs-rpc_pipefs.mount/cgroup.procs
/sys/fs/cgroup//user.slice/user-1000.slice/user@1000.service/dev-mapper-fedora_krowka\x2dswap.swap/cgroup.procs
```

This isn't harmful, but confusing and inefficient. I think we should be able to figure out that the cgroup doesn't need to be realized in those cases.

**Comment by poettering** [comment]
Hmm, so this doesn't happen here (neither on git, nor in f31) , and I can't see how this could happen even... very confusing. Which version are you running?

We generally only realize the cgroup before we fork something off. So either I am very confused, or these units fork something off on your system? Weird...

Either way I fear you have to track this down on your system a bit more...

**Comment by keszybz** [comment]
I still see this. But I'm running some random version (v244-332-gff9e6be124). I'll install the latest git and check.

**Comment by keszybz** [comment]
Yep, I still see this with 244-517-gd21cb43.

For example, `pulseaudio.socket`:
```console
$ cat /sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/pulseaudio.socket/cgroup.procs
(empty)
$ systemctl --user cat pulseaudio.socket
# /usr/lib/systemd/user/pulseaudio.socket
[Unit]
Description=Sound System
ConditionUser=!root

[Socket]
Priority=6
Backlog=5
ListenStream=%t/pulse/native

[Install]
WantedBy=sockets.target
```
So it is pretty clear that nothing was forked to create this socket, and the cgroup shouldn't have been realized.

Looking at the list of units with empty cgroups, those are all either sockets, or "mirror" units for system mounts or swaps.

**Comment by poettering** [comment]
#14554 is supposed to fix this

**Comment by poettering** [comment]
So I am pretty sure I nailed it now, after @w-simon pointed me to the real issue in #14554. Fix in #14564. Could you give this a whirl, please?
